pkg/client/selfhosted/selfhosted.go

```diff
@@ -0,0 +1,254 @@
+	V1Compatibility string `json:"v1Compatibility"`
+}
+
+type V1Compatibility struct {
+	Created time.Time `json:"created,omitempty"`
+}
+
+func New(ctx context.Context, opts Options) (*Client, error) {
+	client := &http.Client{
+		Timeout: time.Second * 5,
+	}
+
+	u, err := url.Parse(opts.URL)
+	if err != nil {
+		// If we can't parse the host given by the options, we should exit fatal
+		log.Fatalf("failed parsing host: %s", opts.URL)
+	}
+
+	// Set the rexex which is used by our path IsHost function
+	opts.HostRegex = fmt.Sprintf(regTemplate, u.Host)
+
+	// Only try to setup auth if an actually URL is present.
+	if opts.URL != "" {
+		// Setup Auth if username and password used.
+		if len(opts.Username) > 0 || len(opts.Password) > 0 {
+			if len(opts.Bearer) > 0 {
+				return nil, errors.New("cannot specify Bearer as well as username/password")
+			}
+
+			token, err := basicAuthSetup(client, opts)
+			if err != nil {
+				return nil, fmt.Errorf("failed to setup auth: %s", err)
+			}
+			opts.Bearer = token
+		}
+	}
+
+	return &Client{
+		Options: opts,
+		Client:  client,
+	}, nil
+}
+
+func (c *Client) Tags(ctx context.Context, _, repo, image string) ([]api.ImageTag, error) {
+	url := fmt.Sprintf(lookupURL, c.Options.URL, repo, image)
+	urlManifest := fmt.Sprintf(manifestURL, c.Options.URL, repo, image)
+	var tags []api.ImageTag
+
+	response, err := c.doRequest(ctx, url)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, tag := range response.Tags {
+
+		manifestResponse, err := c.doManifestRequest(ctx, urlManifest+tag)
+		if err != nil {
+			return nil, err
+		}
+
+		// Set default
+		time := time.Now()
+
+		for _, v1History := range manifestResponse.History {
+			data := V1Compatibility{}
+			if err := json.Unmarshal([]byte(v1History.V1Compatibility), &data); err != nil {
+				return nil, err
+			}
+
+			if &data.Created != nil {
+				time = data.Created
+				// Each layer has its own created timestamp. We just want a general reference.
+				// Take the first and step out the loop
+				break
+			}
+		}
+
+		tags = append(tags, api.ImageTag{
+			Tag:          tag,
+			SHA:          manifestResponse.Digest,
+			Timestamp:    time,
+			Architecture: manifestResponse.Architecture,
+		})
+	}
+
+	return tags, nil
+}
+
+func (c *Client) doManifestRequest(ctx context.Context, url string) (*ManifestResponse, error) {
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req = req.WithContext(ctx)
+	if len(c.Bearer) > 0 {
+		req.Header.Add("Authorization", "Bearer "+c.Bearer)
+	}
+
+	resp, err := c.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get docker image: %s", err)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	response := new(ManifestResponse)
+	if err := json.Unmarshal(body, response); err != nil {
+		return nil, fmt.Errorf("unexpected image tags response: %s", body)
+	}
+
+	response.Digest = resp.Header.Get("Docker-Content-Digest")
```

Comment:
This is only accurate with the header on the request

```
Accept: application/vnd.docker.distribution.manifest.v2+json
```

Reply:
I reverted back to v2.1 because the v2.2: 
- Does not provide the created timestamp
- Is not always present, and will fallback to v1
- v2.1 is still the "existing manifest format" based on https://docs.docker.com/registry/spec/manifest-v2-2/

Both v2.1 and v2.2 return the Header response `Docker-Content-Digest` but v2.1 is richer in the content that we want (created time)

Reply:
Just to be certain, I've added `req.Header.Add("Accept", "application/vnd.docker.distribution.manifest.list.v1+json")` to be sure that for somewhat reason we do not receive the v2.2 version. This should not change anything, but perhaps in the future it would.